Makefile

```diff
@@ -0,0 +1,117 @@
+# Copyright 2019 The Souper Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Invoke with `make deps Build=Debug` or `make deps` (defaults to Release)  
+
+llvm_branch=tags/RELEASE_700/final
+llvm_build_type=$(Build)
+alivedir=third_party/alive2
+alive_builddir=$(alivedir)/build
+hiredis_commit=010756025e8cefd1bc66c6d4ed3b1648ef6f1f95
+klee_repo=https://github.com/rsas/klee
+klee_branch=pure-bv-qf-llvm-7.0
+hiredisdir=third_party/hiredis
+kleedir=third_party/klee
+llvmdir=third_party/llvm
+llvm_installdir=$(llvmdir)/$(llvm_build_type)
+llvm_builddir=$(llvmdir)/${llvm_build_type}-build
+cmake_flags=-DCMAKE_INSTALL_PREFIX="../../../$(llvm_installdir)" -DLLVM_ENABLE_ASSERTIONS=On -DLLVM_TARGETS_TO_BUILD=host -DCMAKE_BUILD_TYPE="$(llvm_build_type)"
+Build=Release
+# $(Build) will not be set here if assigned through the command line
+# TODO: Make the targets build-type specific, currently you have to nuke third_party for changing build type
+NINJA=ninja
+# Override NINJA from the command line with something like NINJA="ninja -j1" if you are running out of memory
+
+deps : alive2 llvm hiredis klee
+
+llvm : third_party/llvm llvm-build
+# 	we want these but they don't get installed by default
+	cp $(llvm_builddir)/bin/llvm-lit $(llvm_installdir)/bin/
+	cp $(llvm_builddir)/bin/FileCheck $(llvm_installdir)/bin/
+	cp $(llvm_builddir)/lib/libgtest_main.a $(llvm_installdir)/lib/
+	cp $(llvm_builddir)/lib/libgtest.a $(llvm_installdir)/lib/
+
+llvm-build:
+	@echo "Building LLVM"
+
+	mkdir -p $(llvm_builddir)
+	
+	if [ -n "`which ninja`" ]; then\
+		$(NINJA) -C $(llvm_builddir);\
+		$(NINJA) -C $(llvm_builddir) install;\
+	else\
+		make -C $(llvm_builddir) -j4;\
+		make -C $(llvm_builddir) -j4 install;\
+	fi
+alive2 : third_party/alive2
```

Comment:
Same .PHONY thing as above. There are other targets in this file as well which are not marked as .PHONY. I guess the problem will start appearing when there are files in the filesystem with exact same name as these targets. `make` won't build these targets because it will think that these targets are already built (even though you want them to be built unconditionally everytime).